### ISSUE TITLE
Uploader not communicating with Contour Link

### ISSUE THREAD

**Issue by nickpeters1975** [pazaan/640gAndroidUploader#67]
- 0.31
- Samsung Galaxy 1 Note
- Android Version Number 4.1.2
- Wi-Fi
- 

I've previously setup Medtronic 640g Nightscout on a Moto G 2 phone, however the OTG connection wasn't good so brought another phone (Samsung Galaxy Note 1) which connection is fine, however when I've gone to install the uploader, I keep getting errors such as "Timeout communicating with the Contour Next Link and one called "Could not close connection: Expected to get control character '6' Got '81'.  I have uploaded the pump data through Carelink prior when i was using it with the Motog 2 and again with this new phone, registered it also, done everything I should have but nothing seems to get it to connect.  The uploader on the MotoG2 still works fine.
#### Uploader Error Code

Timeout communicating with the Contour Next Link
Could not close connection: Expected to get control character '6' Got '81'
Error connecting to Contour Next Link
#### Steps to Reproduce (for bugs)
1.  Install 0.31 APK
2. Register Contour Next Link Successfully
3. Read Data Now
4. Multiple errors.  Yellow light on meter appears to flash and data is going thru wifi on phone.
#### Severity Score

5

**Comment by Pogman** [comment]
Couple of potential things for you to check:

I've seen errors like this on a phone with a poor well worn mini-usb connection. The phone and usb device will connect but due to poor contact on the data pins this error will show. If this is the issue a phone repair place can redo the connector for you.

The phone may not be delivering enough power on the OTG bus. Much less likely but can be checked if you have a OTG cable that takes an external power source.

**Comment by nickpeters1975** [comment]
Apologies for the delay in replying to this thread.

This was the fault, due to a worn Mini USB connection.   Have replaced the phone with a xperia u and unfortunately this phone has begun to have the same issue over time.

Are now using a Galaxy S5 which seems to never lose the connection via the OTG, however I'm still trying to figure out how to keep the uploader running in the background whilst I use the phone in everyday life.  If I keep the screen on using a screen dimmer application (minimum screen brightness) the uploader tends to continue, but can just drop off at anytime (eg 1 or 2 hours if I'm lucky).  Once I access the application by selecting it within the android system (not restarting it), it will appear to begin to connect again.

I notice you use a similar phone (J1)?  Have you found a way to keep the application running in the background?  Ideally I'd like to be able to lock the screen and have it running 24/7 in the background as I have the phone in my pocket unlocked all sorts of funny nonsense can happen!  

Thanks so much.

**Comment by Pogman** [comment]
I'm using a Galaxy S3 - Works OK and I built my own nice and compact OTG connector and superglued it into the phone socket. If I need to I can work it back out but the glue keeps in solid and means I can be quite rough with handling the phone and use it like normal chuck around phone. I added a tiny switch so I could go between OTG and USB modes for charging.

For me the uploader needs to be in focus before the phone auto shuts off the screen. So if I use any other functions or apps on the phone I have to make sure I go back to the uploader before the screen shuts off.

Pogman - aka Stimpy ;)
